std/datetime/timezone.d

```diff
@@ -1699,43 +1699,51 @@ package:
     static immutable(SimpleTimeZone) fromISOExtString(S)(S isoExtString) @safe pure
         if (isSomeString!S)
     {
-        import std.algorithm.searching : startsWith, countUntil, all;
-        import std.ascii : isDigit;
-        import std.conv : to;
+        import std.algorithm.searching : startsWith;
+        import std.conv : ConvException, to;
+        import std.datetime.date : DateTimeException;
         import std.exception : enforce;
         import std.format : format;
+        import std.string : indexOf;
 
-        auto dstr = to!dstring(isoExtString);
-
-        import std.datetime.date : DateTimeException;
-        enforce!DateTimeException(dstr.startsWith('-', '+'), "Invalid ISO String");
-
-        auto sign = dstr.startsWith('-') ? -1 : 1;
-
-        dstr.popFront();
-        enforce!DateTimeException(!dstr.empty, "Invalid ISO String");
+        auto whichSign = isoExtString.startsWith('-', '+');
```

Comment:
`byCodeUnit` would save auto-decoding here.

Reply:
I've already fixed startsWith and endsWith to not auto decode